Don't compute only on non-zero scores

dpsutton opened the pull request:
Fixes #16591 

I was removing a helper function when refactoring the scoring. That
had a filter on :score?. I'm not sure why i translated this to only
interpret the denominator of scoring to only include scores that were
positive. This is obviously wrong.

For the archive page, we just want all archived results, so it is quite reasonable that all scores are 0 (the text input is an empty string), and reducing `+` over those where the score is positive leads us to just `(reduce + ())` and we have a zero in our denominator.

Also, previously, all databases would be returned in the archived
query which seems obviously wrong. Databases cannot be archived and so
should never appear when searching for archived items.

dpsutton committed:
Don't compute only on non-zero scores

I was removing a helper function when refactoring the scoring. That
had a filter on :score?. I'm not sure why i translated this to only
interpret the denominator of scoring to only include scores that were
positive. This is obviously wrong.

Also, previously, all databases would be returned in the archived
query which seems obviously wrong. Databases cannot be archived and so
should never appear when searching for archived items.

dpsutton commented on src/metabase/api/search.clj:
```diff
@@ -163,7 +163,7 @@
 (defmethod archived-where-clause (class Database)
   [model archived?]
-  [:= 1 1])
+  [:= 1 (if archived? 2 1)])
```
@tsmacdonald wanted your eyes specifically on this part. There were databases on the archive page which was quite strange since they cannot be archived. Seems like we would never want databases in archived results

tsmacdonald replied:
Yeah, this seems sensible (thanks for the work!). Some of my early search work was completely unaware of the archived page being powered by the search endpoint (I still think is not the best idea), so when I added databases to the search results I wasn't thinking about it.

dpsutton replied:
I agree. I was quite surprised to see that we were scoring and ranking these things.

tsmacdonald reviewed (approved):
Looks good to me, thanks!

dpsutton committed:
Add some tests

- had been asserting that ee-score = os-score if not in an official
collection. This is actually hard to do under the current scheme, and
actually not important. The important bit is that the relative
ordering is the same. It had been adding 0 score 2 weight previously,
and removing all scores with a score of 0. This created a bad way to
score, as something with 1/2, 0/3, 0/4, 0/10 (net score 1/2=0.5) would
appear higher than something with 1/2, 1/3, 1/4, 3/10 (net score
6/19=0.315) and it really shouldn't. The former really should
be (1/19=0.05). And since we are adding in a denominator of
2 (official collection weight) all of the scores are actually smaller,
but that's ok, as they are all proportionately smaller.
